clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -295,6 +295,47 @@ LogicalResult IntAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DataMemberAttr definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+DataMemberAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                       mlir::cir::DataMemberType ty,
+                       std::optional<size_t> memberIndex) {
+  auto clsStructTy = ty.getClsTy();
+  if (!clsStructTy) {
+    emitError() << "class type of a DataMemberAttr must be a struct type";
+    return failure();
+  }
+
+  if (!memberIndex.has_value()) {
+    // DataMemberAttr without a given index represents a null value.
+    return success();
+  }
+
+  if (clsStructTy.isIncomplete()) {
+    emitError() << "trying to build a non-null ptr to data member of an "
+                   "incomplete class";
```

Comment:
this should probably read like `incomplete 'cir.struct' requires a non-null data member pointer`.